src/Java.Interop.Tools.JavaTypeSystem/JavaModels/IJavaResolvable.cs

```diff
@@ -0,0 +1,10 @@
+using System;
+using System.Collections.Generic;
+
+namespace Java.Interop.Tools.JavaTypeSystem.Models
+{
+	public interface IJavaResolvable
+	{
+		void Resolve (JavaTypeCollection types, List<JavaUnresolvableModel> unresolvables);
```

Comment:
This name is "weird", in that `Resolve` is *already* a "used verb" from Cecil, and in Cecil it returns a value (and takes no parameters).  Is there a better "verb" that describes what would be done?  `Validate()`, perhaps?

Reply:
I think `Resolve` is the correct word because it is doing the same thing as Cecil: converting unparsed data (like a string) into a usable object model:

```csharp
var resolver = new BaseAssemblyResolver ();
var asm = resolver.Resolve ("MyAssembly.dll");

-- or --

var typedef = new TypeDefinition ("System", "String");
var type = typedef.Resolve ();
```

The difference is we are setting a public object model property instead of returning the model.  This is because we are resolving the entire object instead of a single type.  For example, a method may need to resolve the return type and multiple parameter types.

```csharp
public class JavaClassModel : IJavaResolvable
{
  public string BaseType { get; set; }
  public JavaTypeReference BaseTypeModel { get; set; }

  public void Resolve (JavaTypeCollection types, List<JavaUnresolvableModel> unresolvables)
  {
    BaseTypeModel = types.ResolveTyepReference (BaseType);
  }
}
```

I feel like `Validate` implies merely checking if the data is valid, and not converting it to a usable object model.